### PR TITLE
Feature/register vm

### PR DESCRIPTION
**Pull request by nikilark**
Now we can register a new user!
Main feature is possibility to upload photo. Though, to do this you need to enter your student card number
The picture will store in chosen directory (now it Widgets/Images/Profile_Images) and have name equal to your student card to make them unique

### CODE REVIEW

File: View/Pages/RegisterPage.xaml.cs [hunk 1]
```diff
@@ -20,12 +23,52 @@ namespace View.Pages
...
             navbar.button_login.Click += LogInNavbar_Buttom_login_Click;
             navbar.button_FAQ.Click += LogInNavbar_Button_FAQ_Click;
+            content.button_cancel.Click += Content_Button_cancel_Click;
+            content.button_upload_photo.Click += Content_Button_upload_photo_Click;
+            content.button_Register.Click += Content_Button_Register_Click;
+        }
+
+        private void Content_Button_Register_Click(object sender, RoutedEventArgs e)
+        {
+            if (logic.validateValues(content.edit_Name.Text, content.edit_Surname.Text, content.edit_Zal.Text, content.edit_Username.Text, content.edit_Password.Password, content.edit_RepPassword.Password))
```
Comment (RomanLevkovych) [inline View/Pages/RegisterPage.xaml.cs]: Simplify

File: View/Pages/RegisterPage.xaml.cs [hunk 2]
```diff
@@ -20,12 +23,52 @@ namespace View.Pages
     /// </summary>
     public partial class RegisterPage : Page
     {
+        private RegisterPageVM logic;
```
Comment (RomanLevkovych) [inline View/Pages/RegisterPage.xaml.cs]: Add more meaningful name to the field

File: ViewModel/RegisterPageVM.cs [hunk 3]
```diff
@@ -1,10 +1,63 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Text;
+using System.IO;
+using DataServices.Services;
 
 namespace ViewModel
 {
-    class RegisterPageVM
```
Comment (RomanLevkovych) [inline ViewModel/RegisterPageVM.cs]: Add more meaningful name basing on the class responsibilities.

Comment (Scaler358) [reply ViewModel/RegisterPageVM.cs]: But we call all ViewModel classes like that.
<joke>
(ValidatorCopiratorRegistratorDBConnectorForRegisterPage?)
</joke>

File: View/Pages/RegisterPage.xaml.cs [hunk 4]
```diff
@@ -36,11 +36,16 @@ public RegisterPage()
             content.button_Register.Click += Content_Button_Register_Click;
         }
 
+        private bool validateValues()
+        {
+            return viewModel.validateValues(content.edit_Name.Text, content.edit_Surname.Text, content.edit_Zal.Text, content.edit_Username.Text, content.edit_Password.Password, content.edit_RepPassword.Password);
+        }
+
         private void Content_Button_Register_Click(object sender, RoutedEventArgs e)
```
Comment (OlegAndrus12) [inline View/Pages/RegisterPage.xaml.cs]: Use meaningful name, not the generated one for buttons actions.
+camelCase

File: View/Pages/RegisterPage.xaml.cs [hunk 5]
```diff
@@ -20,12 +23,52 @@ namespace View.Pages
     /// </summary>
     public partial class RegisterPage : Page
     {
+        private RegisterPageVM logic;
         public RegisterPage()
         {
+            logic = new RegisterPageVM(new SQLiteDataService());
             InitializeComponent();
             navbar.button_register.Click += LogInNavbar_Button_register_Click;
```
[the remark this answers is not in the document]
Comment (OlegAndrus12) [reply View/Pages/RegisterPage.xaml.cs]: RegisterPageVM represent class responsibility correctly, as for me

**Review by OlegAndrus12 — CHANGES_REQUESTED** [review]

File: View/Widgets/RegisterPageContent.xaml [hunk 6]
```diff
@@ -43,8 +43,8 @@
         <TextBox x:Name="edit_Surname" Grid.Row="2" Grid.Column="5" Height="25" FontSize="18"></TextBox>
```
Comment (OlegAndrus12) [inline View/Widgets/RegisterPageContent.xaml]: Camel

File: View/Pages/RegisterPage.xaml.cs (hunk 4 above)
Comment (Scaler358) [reply View/Pages/RegisterPage.xaml.cs]: Can you look at this method and say what it doing? If yes, it is meaningful name

**Review by OlegAndrus12 — APPROVED** [review]

**Pull request merged**